[WIP]Send show_jid output to stderr

tacerus opened the pull request:
### What does this PR do?

When using --out=json, all data sent to stdout should be valid JSON to allow for piping into processing tools such as `jq`. Make this the case with "show_jid" being enabled as well, by printing the JID line to stderr instead of stdout.

### What issues does this PR fix or reference?
Fixes https://github.com/saltstack/salt/issues/25154 (the issue was closed with `--static` being suggested as a solution, however this to me is merely a workaround, as it makes for a different output strategy).

### Previous Behavior

With `show_jid` enabled, piping into JSON processing tools leads to parsing exceptions.

### New Behavior

stdout can be processed as JSON with `show_jid` enabled, the same way as with `show_jid` disabled.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

<!-- Please review Salt's Contributing Guide for best practices and guidance in
choosing the right branch:
https://docs.saltproject.io/en/master/topics/development/contributing.html -->

<!-- Additional guidance for pull requests can be found here:
https://docs.saltproject.io/en/master/topics/development/pull_requests.html -->

<!-- See GitHub's page on GPG signing for more information about signing commits
with GPG:
https://help.github.com/articles/signing-commits-using-gpg/ -->

tacerus committed:
Send show_jid output to stderr

When using --out=json, all data sent to stdout should be valid JSON to
allow for piping into processing tools such as `jq`.
Make this the case with "show_jid" being enabled as well, by printing
the JID line to stderr instead of stdout.

Signed-off-by: Georg Pfuetzenreuter <mail@georg-pfuetzenreuter.net>

tacerus commented:
I tried adding tests with

```
salt_cli.run("test.ping --out=json", minion_tgt=salt_minion.id)
```

in tests/pytests/integration/cli/test_salt.py and also attempted a few variations of placing the `out` option, but it seems none of these work, as `--out=json` is already implicitly added in `salt_cli.run()`:

```
E           ProcessResult
E            Command Line: ['/home/georg/Work/git/salt/.nox/test-parametrized-3-crypto-none-transport-zeromq-coverage-false/bin/python', '/tmp/stsuite/scripts/c
li_salt.py', '--config-dir=/tmp/stsuite/master-GUfyDf/conf', '--timeout=30', '--out=json', '--out-indent=0', '--log-level=critical', 'minion-dCDa3r', 'test.ping
 --out=json']
```

Can I get regular / unfiltered CLI output in the test suite somehow?

tacerus commented:
Hi @mattp-, thanks for your input and the better solution - I will test, but from briefly reading over your patch, it seems it would cause the jid to not be printed at all if operating with the json or yaml output mode? I understand why, however think it would still be useful to have it available - maybe at least for JSON mode, should we print it as a separate JSON object?

mattp- commented:
no, it would. implicitly the jid is in the 'full_return'/raw load returned over the wire; i should have pasted an output example:
```
mphillips81@mmp-pw-dev /mnt/dev/repos/upstream-salt (sqla)✗ % uv run salt local -l info test.ping  --out=json --show-jid 2>/dev/null
{
    "local": {
        "ret": true,
        "retcode": 0,
        "jid": "20250604233927813243"
    }
}
```

dwoz committed:
Add tests verifying show_jid output goes to stderr

Test all three get_cli_* methods and update the existing functional
test to assert jid: line appears on stderr, not stdout, keeping
--out=json stdout valid JSON. Fixes #25154.
